tests/templates/test_embeddings.py

```diff
@@ -329,6 +329,21 @@ def circuit(x=None):
         with pytest.raises(ValueError, match="'basis_state' must only consist of"):
             circuit(x=[2, 3])
 
```

Comment:
Also snuck in this test of the additional check in ``BasisEmbedding``

Reply:
Yes, it is the same thing, just in a different packaging and with different input checks :) But not all users may know this, so the templates may be used in other contexts.

I could make one dependent on the other, but find it more elegant to implement the 2 lines of code from scratch, for better maintainability.

``BasisState`` "hacks" the state preparation on simulators as far as I know.